pims/base_frames.py

```diff
@@ -133,45 +133,145 @@ def __repr__(self):
                                   dtype=self.pixel_type)
 
 
-class FramesSequence(FramesStream):
-    """Baseclass for wrapping data buckets that have random access.
+class SliceableIterable(object):
```

Comment:
The class name doesn't match with the import in api.py. I think renaming this to `SliceableIterator` is the best option, this also goes to multiple instances below.